### PR TITLE
ref(tsc): Convert ProjectCspReport to FC and useApiQuery

### PR DESCRIPTION
**Pull request by ArthurKnaus**
Convert to FC and useApiQuery.
Use `useOrganization` and `useParams` instead of receiving them via props.

- Ref https://github.com/getsentry/frontend-tsc/issues/2

### CODE REVIEW

File: static/app/views/settings/projectSecurityHeaders/csp.tsx [hunk 1]
```diff
@@ -8,76 +6,68 @@ import Panel from 'sentry/components/panels/panel';
...
+    [`/projects/${organization.slug}/${projectId}/keys/`],
+    {staleTime: 0}
+  );
+  const {data: project} = useApiQuery<Project>(
+    [`/projects/${organization.slug}/${projectId}/`],
+    {staleTime: 0}
+  );
+
+  if (!keyList || !project) {
+    return null;
```
Comment (scttcper) [inline static/app/views/settings/projectSecurityHeaders/csp.tsx:64]: For this to be similar to async view that its replacing there should be a loading state and an error state

File: static/app/views/settings/projectSecurityHeaders/csp.tsx [hunk 2]
```diff
@@ -8,76 +6,68 @@ import Panel from 'sentry/components/panels/panel';
...
+  const projectId = params.projectId;
+
+  const {data: keyList} = useApiQuery<ProjectKey[]>(
+    [`/projects/${organization.slug}/${projectId}/keys/`],
+    {staleTime: 0}
+  );
+  const {data: project} = useApiQuery<Project>(
+    [`/projects/${organization.slug}/${projectId}/`],
+    {staleTime: 0}
+  );
```
Comment (scttcper) [inline static/app/views/settings/projectSecurityHeaders/csp.tsx:61]: is this even used?

File: static/app/views/settings/projectSecurityHeaders/csp.tsx (hunk 1 above)
Comment (ArthurKnaus) [reply static/app/views/settings/projectSecurityHeaders/csp.tsx:64]: yes, will add 👍 
I see the AsyncComponent has quite a bit of logic for the error handling.
Do we have a helper for that or is not necessary to do all that nowadays?

File: static/app/views/settings/projectSecurityHeaders/csp.tsx (hunk 2 above)
Comment (ArthurKnaus) [reply static/app/views/settings/projectSecurityHeaders/csp.tsx:61]: great find! trusted the previous implementation too much 🙈

Comment (ArthurKnaus) [reply static/app/views/settings/projectSecurityHeaders/csp.tsx:61]: The from is pre-filled with `initialData={project.options}`

File: static/app/views/settings/projectSecurityHeaders/csp.tsx (hunk 1 above)
Comment (scttcper) [reply static/app/views/settings/projectSecurityHeaders/csp.tsx:64]: i've been hooking up <LoadingError with the retry calling refetch and that's about it. There's still error handling at the router level

Comment (scttcper) [reply static/app/views/settings/projectSecurityHeaders/csp.tsx:64]: https://www.notion.so/sentry/Guide-How-to-convert-AsyncComponent-and-AsyncView-to-useApiQuery-55751df48dc04df5b4b0da42ba5e4fbd?pvs=4#e771950dabea4588bb1f8ee555d67214

**Review by scttcper — APPROVED** [review]

**Pull request merged**